hazelcast-client/src/main/java/com/hazelcast/client/impl/ClientLockReferenceCounter.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.client.impl;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public final class ClientLockReferenceCounter {
```

Comment:
`...ReferenceCounter` name looks misleading. It doesn't count references, it's just creating/generating reference-id. Something like generator, factory...